manager/controllers/app/select_modules.go

```diff
@@ -115,8 +128,11 @@ func (p *PlotterGenerator) validate(item *DataInfo, solution Solution, applicati
 		element := &solution.DataPath[ind]
 		element.Actions = []taxonomy.Action{}
 		moduleCapability := element.Module.Spec.Capabilities[element.CapabilityIndex]
-		if !element.Edge.Sink.Virtual {
+		if !element.Edge.Sink.Virtual &&
+			(item.Context.Flow != taxonomy.WriteFlow ||
+				(item.Context.Flow == taxonomy.WriteFlow && item.Context.Requirements.FlowParams.IsNewDataSet)) {
```

Comment:
```
(item.Context.Flow != taxonomy.WriteFlow ||
				(item.Context.Flow == taxonomy.WriteFlow && item.Context.Requirements.FlowParams.IsNewDataSet))
```
1. This logic appears in several places. Create a function that based on `item` returns true if storage is needed to be allocated.
2. Simplify. The same logic can be expressed as 
`(item.Context.Flow != taxonomy.WriteFlow || item.Context.Requirements.FlowParams.IsNewDataSet)`